Bug: Intra-Crate Integration Tests panic with TargetNotInstalled in runserver_hot_reload

## Summary

Two hot-reload integration tests in `crates/reinhardt-commands/tests/runserver_hot_reload.rs` panic during the **Intra-Crate Integration Tests** CI job because the `wasm32-unknown-unknown` target is not installed on the runner before the tests invoke an in-process wasm build.

## Failing Tests

| Shard | Test | Location |
|---|---|---|
| `Intra-Crate Integration Tests (2/8)` | `hr_1_wasm_change_triggers_wasm_rebuild` | `crates/reinhardt-commands/tests/runserver_hot_reload.rs:240:38` |
| `Intra-Crate Integration Tests (5/8)` | `hr_6_issue_4128_reproduction` | `crates/reinhardt-commands/tests/runserver_hot_reload.rs:422:38` |

Both fail with the same panic:

```
thread '<test>' panicked at crates/reinhardt-commands/tests/runserver_hot_reload.rs:<line>:38:
initial wasm build must succeed: TargetNotInstalled
```

The `.expect("initial wasm build must succeed")` call unwraps a `cargo` invocation that ultimately reports `TargetNotInstalled` for `wasm32-unknown-unknown`.

## Reproduction

Reproduces on every recent CI run on `main` (and on every open PR that does not modify the workflow), e.g.:

- Run: <https://github.com/kent8192/reinhardt-web/actions/runs/25431079973>
  - Job 2/8: <https://github.com/kent8192/reinhardt-web/actions/runs/25431079973/job/74606723102>
  - Job 5/8: <https://github.com/kent8192/reinhardt-web/actions/runs/25431079973/job/74606723116>

Failure summary from the run:

```
Summary [83.401s] 614 tests run: 613 passed, 1 failed, 4370 skipped
error: test run failed
##[error]Process completed with exit code 100.
```

## Expected Behavior

Both `hr_1_*` and `hr_6_*` should execute their hot-reload scenarios without panicking. The CI environment must have `wasm32-unknown-unknown` installed before these tests run, or the tests must install/skip when the target is missing.

## Actual Behavior

`TargetNotInstalled` panic causes shard 2/8 and 5/8 to fail, blocking the **Intra-Crate Integration Tests** umbrella check on every PR.

## Suggested Fixes (pick one)

1. **Workflow fix (preferred for CI parity)** — add `rustup target add wasm32-unknown-unknown` to the `Intra-Crate Integration Tests` job in `.github/workflows/` before `cargo nextest`. The recently-merged PR #4181 added `wasm-bindgen-cli` install but did not add the target itself.
2. **Test-side fix** — call `rustup target add wasm32-unknown-unknown` (or assert availability with a clearer skip message) in the test's setup helper so a missing target is handled gracefully instead of panicking from `.expect(...)`.

Option (1) is recommended because the tests are explicitly designed to exercise the wasm build path; silently skipping would erode coverage.

## Environment

- Runner: GitHub-hosted Linux x64 (ubuntu-latest)
- Toolchain: rust 1.94.1 (from `rust-toolchain.toml`)
- Affected shards: 2/8 and 5/8 of `Intra-Crate Integration Tests`
- First observed: 2026-05-06 runs (correlates with PR #4181 wasm-bindgen-cli install change in window)

## Related

- PR #4181 — added `wasm-bindgen-cli` install but apparently did not add the target itself
- Issue tracking the wider WASM CI gate: #4161

## Verification Steps

- [ ] Add `rustup target add wasm32-unknown-unknown` to the intra-crate workflow (or equivalent test-side fix)
- [ ] Re-run `Intra-Crate Integration Tests` on a PR and confirm shards 2/8 and 5/8 pass
- [ ] Confirm `hr_1_wasm_change_triggers_wasm_rebuild` and `hr_6_issue_4128_reproduction` complete with `passed`

--- a/.github/workflows/intra-crate-integration-test.yml
+++ b/.github/workflows/intra-crate-integration-test.yml
@@ -128,6 +128,14 @@ jobs:
         with:
           tool: wasm-bindgen-cli
 
+      # Required by hot-reload tests in crates/reinhardt-commands
+      # (runserver_hot_reload.rs::hr_1_wasm_change_triggers_wasm_rebuild,
+      # hr_6_issue_4128_reproduction) which invoke an in-process
+      # `cargo build --target wasm32-unknown-unknown` and panic with
+      # TargetNotInstalled when the target is missing. Tracked in #4192.
+      - name: Install wasm32-unknown-unknown target
+        run: rustup target add wasm32-unknown-unknown
+
       # Scope mold to the host target only. A global `RUSTFLAGS=...-fuse-ld=mold`
       # leaks into wasm32 builds (e.g. wasm-pack fixtures) and clang rejects
       # `-fuse-ld=mold` for that target. See reinhardt-web#4147.

